stdlib/src/builtin/float_literal.mojo

```diff
@@ -250,18 +250,26 @@ struct FloatLiteral(
         if not self._is_normal():
             return self
 
+        alias one = __mlir_attr.`#kgen.int_literal<1> : !kgen.int_literal`
+        alias neg_one = __mlir_attr.`#kgen.int_literal<-1> : !kgen.int_literal`
         var truncated: IntLiteral = self.__int_literal__()
         var result: Self
-        if abs(self) - abs(truncated) <= 0.5:
-            result = Self(truncated)
-        elif self > 0:
-            result = Self(truncated + 1)
+        var abs_diff = abs(self - truncated)
+        var plus_one = one if self > 0 else neg_one
+        if abs_diff == 0.5:
+            # Round to the nearest even number.
+            if truncated % 2 == 0:
+                result = Self(truncated)
+            else:
+                result = Self(truncated + plus_one)
+        elif abs_diff > 0.5:
+            result = Self(truncated + plus_one)
```

Comment:
here and below: It seems result is only set once, then returned. Can you just early return?
```suggestion
                return Self(truncated)
            else:
                return Self(truncated + plus_one)
        elif abs_diff > 0.5:
            return Self(truncated + plus_one)
```

Reply:
Good idea! Done: https://github.com/msaelices/mojo/commit/5b61c0cab6af500cc0b8f2d8f61ce1c05cee2009